ex-4-quicksort/ex-4-1-recursive-sum.js

```diff
@@ -0,0 +1,12 @@
+//Exercise 4.1
+//Sum all elements in an array with a recursive function
+
+function sum( arr ) {
+    if ( arr.length <= 0) {
+        return 0;
+    }
+
+    return arr[0] + sum([...arr.slice(1, arr.length)]);
```

Comment:
Is it necessary to spread the result of `slice` into this new array?

Reply:
No!  Thanks for pointing that out.